spec/javascripts/stimulus/tabs_and_panels_controller_test.js

```diff
@@ -26,23 +36,75 @@ describe("EnterprisePanelController", () => {
         </div>`;
     });
 
-    it("displays only the default panel", () => {
-      const peekPanel = document.getElementById("peek_panel");
-      const kaPanel = document.getElementById("ka_panel");
-      const booPanel = document.getElementById("boo_panel");
-
-      expect(peekPanel.style.display).toBe("block");
-      expect(kaPanel.style.display).toBe("none");
-      expect(booPanel.style.display).toBe("none");
+    it('displays only the default panel', () => {
+      checkDefaultPanel()
     });
 
-    it("displays appropriate panel when associated tab is clicked", () => {
-      const kaPanel = document.getElementById("ka_panel");
-      const ka = document.getElementById("ka");
+    describe('when tab is clicked', () => {
+      let ka;
 
-      expect(kaPanel.style.display).toBe("none");
-      ka.click();
-      expect(kaPanel.style.display).toBe("block");
-    });
+      beforeEach(() => {
+        ka = document.getElementById('ka');
+      })
+
+      it('displays appropriate panel', () => {
+        const kaPanel = document.getElementById('ka_panel');
+
+        expect(kaPanel.style.display).toBe('none');
+        ka.click();
+        expect(kaPanel.style.display).toBe('block');
+      });
+
+      it('selects the clicked tab', () => {
+        ka.click();
+        expect(ka.classList.contains('selected')).toBe(true);
+      });
+    })
+
+    describe('when anchor is specified in the url', () => {
+      const { location } = window;
+      const mockLocationToString = (panel) => {
+        // Mocking window.location.toString() 
+        const url = `http://localhost:3000/admin/enterprises/great-shop/edit#!#${panel}`
```

Comment:
Note that this might be affected by [this PR](https://github.com/openfoodfoundation/openfoodnetwork/pull/9729): 
> Remove #!# from URLs as they aren't valid.

Reply:
This isn't a problem. 
The URL would be come something like `http://localhost:3000/admin/enterprises/great-shop/edit#${panel}`
and the following code will picks up the string after the last `#` :

https://github.com/rioug/openfoodnetwork/blob/b0462983d747cbd5d86839df047262f87108983f/app/webpacker/controllers/tabs_and_panels_controller.js#L17-L18

Reply:
@dacook turned out you were right, the PR did break my original solution :disappointed: